circleguard/comparer.py

```diff
@@ -103,15 +108,69 @@ def _result(self, replay1, replay2):
             The result of comparing ``replay1`` to ``replay2``.
         """
         self.log.log(utils.TRACE, "comparing %r and %r", replay1, replay2)
-        result = Comparer._compare_two_replays(replay1, replay2)
-        mean = result[0]
-        sigma = result[1]
+
+        if CleanMode.SEARCH in self.detect.clean_mode:
+            mean = Comparer._compare_hill_climb(replay1, replay2, self.detect.clean_mode)
+        else:
```

Comment:
It seems then that there are really two distinct options in `CleanMode` - preprocessing (`VALIDATE`, `SYNCHRONIZE`, `ALIGN`) and processing (actual comparison) (`SEARCH`). Unless I'm misunderstanding that, using a single variable - `clean_mode` - for both of them is very confusing.

Reply:
There are cases in which invalid data is stolen so using only `SYNCHRONIZE + ALIGN` is wanted, some framerate mismatches may cause `SYNCHRONIZE` to behave weirdly and only interpolating between two replays is preferable. I do agree that `VALIDATE + SEARCH` could be mutually exclusive to `SYNCHRONIZE + ALIGN`.